Tribler/Main/tribler_main.py

```diff
@@ -216,7 +211,7 @@ def __init__(self, params, installdir, autoload_discovery=True,
 
             # Schedule task for checkpointing Session, to avoid hash checks after
             # crashes.
-            self.guiserver.add_task(self.guiservthread_checkpoint_timer, SESSION_CHECKPOINT_INTERVAL)
+            session.lm.rawserver.call_in_thread(SESSION_CHECKPOINT_INTERVAL, self.guiservthread_checkpoint_timer)
```

Comment:
In the gui I would use the start_worker stuff and then specify that it should run in the threadpool.